src/emailservice/email_server.py

```diff
@@ -143,7 +144,38 @@ def start(dummy_mode):
   except KeyboardInterrupt:
     server.stop(0)
 
+def initStackdriverProfiling():
+  enable_profiler = None
+  project_id = None
+  try:
+    enable_profiler = os.environ["ENABLE_PROFILER"]
+    project_id = os.environ["GCP_PROJECT_ID"]
```

Comment:
do not require `GCP_PROJECT_ID`. 
users will never set this.

Reply:
It is not required. If it is provided, it will be used. If its not, start() will try to get it from metadata engine.

Reply:
Doesn’t it throw KeyError when not specified?
Why do you do try...except here, when you can do os.getenv(“Key”, None) instead?

Reply:
No specific reason, I'd attribute it to personal preferences. Is there a reason not to use try...except clause in this case?

Reply:
Just more idiomatic. os.getenv exists so using it would be better, but it's fine for now.